src/shared/lib/fetch/index.js

```diff
@@ -270,8 +270,44 @@ export const getArcGISCSVURLFromOrgId = async function(serverNumber, orgId, laye
  * @param {*} dashboardId the ID of the dashboard, as passed to the iframe that renders it (i.e. https://maps.arcgis.com/apps/opsdashboard/index.html#/ec4bffd48f7e495182226eee7962b422 is dashboardId = ec4bffd48f7e495182226eee7962b422)
  * @param {*} layerName the name of the layer to fetch data for, find this by examining requests
  */
-export const getArcGISCSVURL = async function(serverNumber, dashboardId, layerName) {
+export const getArcGISCSVURL = async (serverNumber, dashboardId, layerName) => {
   const dashboardManifest = await json(`https://maps.arcgis.com/sharing/rest/content/items/${dashboardId}?f=json`);
   const { orgId } = dashboardManifest;
   return getArcGISCSVURLFromOrgId(serverNumber, orgId, layerName);
 };
+
+/**
+ * Retrieves data from an ArcGIS REST API. By default, it will retrieve all items at the provided linked with geometry turned off.
+ * You can control pagination size through the `k` parameter in `options`
+ * @param {string} url URL of the resource
+ * @param {*} date the date associated with this resource, or false if a timeseries data
+ * @param {object} options customizable options:
+ *  - k: number of features we want to receive for each request. A smaller number means more request to grab the complete dataset,
+ *  a larger number may result in a partial dataset if we request more than `Max Record Count`. Defaults to 500.
+ *  - additionalParams: additional parameters for this request. Defaults to `where=0%3D0&outFields=*&returnGeometry=false`.
+ *  - alwaysRun: fetches from URL even if resource is in cache, defaults to false
+ *  - disableSSL: disables SSL verification for this resource, should be avoided
+ */
+export const arcGISjson = async (featureLayerURL, date, options = {}) => {
+  const { k, additionalParams } = {
+    k: 500,
+    additionalParams: 'where=0%3D0&outFields=*&returnGeometry=false',
+    ...options
+  };
+
+  const url = `${featureLayerURL.replace(/\?.*$/, '')}?f=json${additionalParams ? `&${additionalParams}` : ''}`;
+
+  const output = [];
+
+  let n = 0;
+  let response = await json(`${url}&resultOffset=${n}&resultRecordCount=${n + k}`, date, options);
+
+  while (response && response.features.length > 0) {
+    n += k;
```

Comment:
```suggestion
    n += response.features.length;
```

So that if our choice of `k` exceeds their options, we don't skip records.

Reply:
@jzohrab regarding `cachekey`: 

Let's say we have a source with 603 cases. We initially set `k=1000`. And let's say the source has limited queries to return a maximum of 500 records.
First fetch will be `resultOffset = 0; resultRecordCount = 1000`, and receive 500 records.
Second fetch will be `resultOffset = 500; resultRecordCount = 1000`, and receive 103 records.
Third fetch will be `resultOffset = 603; resultRecordCount = 1000`, and receive 0 records.
So on that day, we will cache three files, provided that `cachekey` contains `resultOffset`'s value in it.

Let's say at the next scrape date, there are 1203 cases. You can see immediately that `resultOffset` will be different for the third fetch, and thus its `cachekey` would change. BUT, this is a new scrape date, so it should be fetching everything from the server anyway, right?

Reply:
@shaperilio - yes, that's right.  New scrape date won't hit the cache.

This method will work with the caching in the current system, b/c the URL is encoded to a hash.

Doesn't the result set contain something like a "nextPageToken" or something for the next request?  It looks like this just keeps fetching with offsets and recordcounts.  That will also work with caching, no worries. :-)